src/assets/wise5/components/draw/draw-student/draw-student.component.ts

```diff
@@ -0,0 +1,437 @@
+      this.setAuthoredBackgroundAfterTimeout();
+    }
+  }
+
+  isStarterDrawDataAvailable(): boolean {
+    return this.componentContent.starterDrawData != null;
+  }
+
+  isAuthoredBackgroundAvailable(): boolean {
+    return this.componentContent.background != null && this.componentContent.background !== '';
+  }
+
+  setAuthoredBackgroundIfAvailable(): void {
+    if (this.isAuthoredBackgroundAvailable()) {
+      this.setBackgroundImage(this.componentContent.background);
+    }
+  }
+
+  setBackgroundImage(image: string): void {
+    this.drawingTool.setBackgroundImage(image);
+  }
+
+  setAuthoredBackgroundAfterTimeout(): void {
+    // set the background after a timeout to make sure it gets set after the draw data is set.
+    // if we don't do this, the background may not overwrite the existing background that is already
+    // in the draw data.
+    setTimeout(() => {
+      this.setAuthoredBackgroundIfAvailable();
+    }, 500);
+  }
+
+  setDrawingChangedListener(): void {
+    /*
+     * Wait before we start listening for the drawing:changed event. We need to wait
+     * because the calls above to this.drawingTool.setBackgroundImage() will cause
+     * the drawing:changed event to be fired from the drawingTool, but when that happens,
+     * we don't want to call this.studentDataChanged() because it marks the student work
+     * as dirty. We only want to call this.studentDataChanged() when the drawing:changed
+     * event occurs in response to the student changing the drawing and this timeout
+     * will help make sure of that.
+     */
+    setTimeout(() => {
+      this.drawingTool.on('drawing:changed', () => {
+        this.studentDataChanged();
+      });
+    }, 500);
+  }
+
+  setToolChangedListener(): void {
+    this.drawingTool.on('tool:changed', (toolName: string) => {
+      const category = 'Tool';
+      const event = 'toolSelected';
+      const data = {
+        selectedToolName: toolName
+      };
+      this.StudentDataService.saveComponentEvent(this, category, event, data);
+    });
+  }
+
+  handleConnectedComponentsPostProcess(): void {
+    this.setAuthoredBackgroundIfAvailable();
+  }
+
+  /**
+   * Setup the tools that we will make available to the student.
+   */
+  setUpTools(): void {
+    this.DrawService.setUpTools(this.drawingToolId, this.componentContent.tools, this.isDisabled);
+  }
+
+  setStudentWork(componentState: any): void {
+    this.setDrawData(componentState);
+    this.processLatestStudentWork();
+  }
+
+  resetDrawing(): void {
+    if (confirm($localize`Are you sure you want to clear your drawing?`)) {
+      this.drawingTool.clear();
+      if (this.UtilService.hasConnectedComponent(this.componentContent)) {
+        this.handleConnectedComponents();
+      } else if (this.componentContent.starterDrawData != null) {
+        this.drawingTool.load(this.componentContent.starterDrawData);
+      }
+      this.setAuthoredBackgroundIfAvailable();
```

Comment:
It looks like this code block duplicates parts of ```initializeStudentData()``` Can we extract to common function?